pkg/services/ngalert/api.go

```diff
@@ -0,0 +1,183 @@
+		alertDefinitions.Post("/", middleware.ReqSignedIn, binding.Bind(SaveAlertDefinitionCommand{}), api.Wrap(ng.CreateAlertDefinitionEndpoint))
+		alertDefinitions.Put("/:alertDefinitionId", validateOrgAlertDefinition, binding.Bind(UpdateAlertDefinitionCommand{}), api.Wrap(ng.UpdateAlertDefinitionEndpoint))
+	})
+}
+
+// ConditionEval handles POST /api/alert-definitions/eval.
+func (ng *AlertNG) ConditionEval(c *models.ReqContext, dto EvalAlertConditionCommand) api.Response {
+	alertCtx, cancelFn := context.WithTimeout(context.Background(), setting.AlertingEvaluationTimeout)
+	defer cancelFn()
+
+	alertExecCtx := eval.AlertExecCtx{Ctx: alertCtx, SignedInUser: c.SignedInUser}
+
+	fromStr := c.Query("from")
+	if fromStr == "" {
+		fromStr = "now-3h"
+	}
+
+	toStr := c.Query("to")
+	if toStr == "" {
+		toStr = "now"
+	}
+
+	execResult, err := dto.Condition.Execute(alertExecCtx, fromStr, toStr)
+	if err != nil {
+		return api.Error(400, "Failed to execute conditions", err)
+	}
+
+	evalResults, err := eval.EvaluateExecutionResult(execResult)
+	if err != nil {
+		return api.Error(400, "Failed to evaluate results", err)
+	}
+
+	frame := evalResults.AsDataFrame()
+	df := tsdb.NewDecodedDataFrames([]*data.Frame{&frame})
+	instances, err := df.Encoded()
+	if err != nil {
+		return api.Error(400, "Failed to encode result dataframes", err)
+	}
+
+	return api.JSON(200, util.DynMap{
+		"instances": instances,
+	})
+}
+
+// §AlertDefinitionEval handles GET /api/alert-definitions/eval/:dashboardId/:panelId/:refId".
+func (ng *AlertNG) AlertDefinitionEval(c *models.ReqContext) api.Response {
+	alertDefinitionID := c.ParamsInt64(":alertDefinitionId")
+
+	fromStr := c.Query("from")
+	if fromStr == "" {
+		fromStr = "now-3h"
+	}
+
+	toStr := c.Query("to")
+	if toStr == "" {
+		toStr = "now"
+	}
+
+	conditions, err := ng.LoadAlertCondition(alertDefinitionID, c.SignedInUser, c.SkipCache)
+	if err != nil {
+		return api.Error(400, "Failed to load conditions", err)
+	}
+
+	alertCtx, cancelFn := context.WithTimeout(context.Background(), setting.AlertingEvaluationTimeout)
+	defer cancelFn()
+
+	alertExecCtx := eval.AlertExecCtx{Ctx: alertCtx, SignedInUser: c.SignedInUser}
+
+	execResult, err := conditions.Execute(alertExecCtx, fromStr, toStr)
+	if err != nil {
+		return api.Error(400, "Failed to execute conditions", err)
+	}
+
+	evalResults, err := eval.EvaluateExecutionResult(execResult)
+	if err != nil {
+		return api.Error(400, "Failed to evaluate results", err)
+	}
+
+	frame := evalResults.AsDataFrame()
+
+	df := tsdb.NewDecodedDataFrames([]*data.Frame{&frame})
+	instances, err := df.Encoded()
+	if err != nil {
+		return api.Error(400, "Failed to encode result dataframes", err)
+	}
+
+	return api.JSON(200, util.DynMap{
+		"instances": instances,
+	})
+}
+
+// GetAlertDefinitionEndpoint handles GET /api/alert-definitions/:alertDefinitionId.
+func (ng *AlertNG) GetAlertDefinitionEndpoint(c *models.ReqContext) api.Response {
```

Comment:
Doesn't have to be public:
```suggestion
// getAlertDefinitionEndpoint handles GET /api/alert-definitions/:alertDefinitionId.
func (ng *AlertNG) getAlertDefinitionEndpoint(c *models.ReqContext) api.Response {
```